yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1025,9 +1025,6 @@ object Client extends Logging {
         "future version of Spark. Use ./bin/spark-submit with \"--master yarn\"")
     }
 
-    // Set an env variable indicating we are running in YARN mode.
-    // Note that any env variable with the SPARK_ prefix gets propagated to all (remote) processes
-    System.setProperty("SPARK_YARN_MODE", "true")
```

Comment:
to @tgravescs point, would it hurt to leave this in here?  and then change the comment to indicate that it should normally be set by SparkSubmit but left here in case anybody is still using this.